app/assets/javascripts/admin/addon/components/theme-translation.js

```diff
@@ -1,16 +1,23 @@
 import SiteSettingComponent from "./site-setting";
+import { ajax } from "discourse/lib/ajax";
+import { url } from "discourse/lib/computed";
 import { alias } from "@ember/object/computed";
 
-export default class ThemeTranslation extends SiteSettingComponent {
+export default class extends SiteSettingComponent {
   @alias("translation") setting;
   @alias("translation.key") settingName;
+  @url("model.id", "/admin/themes/%@") updateUrl;
 
   type = "string";
 
   _save() {
-    return this.model.saveTranslation(
-      this.get("translation.key"),
-      this.get("buffered.value")
-    );
+    const translations = {
+      [this.get("translation.key")]: this.get("buffered.value"),
+    };
+
+    return ajax(this.updateUrl, {
+      type: "PUT",
+      data: { theme: translations },
+    });
```

Comment:
Don't we want to update the model object locally too?

Reply:
No, we don't, because the model updating is the issue currently. The model here is the theme, it includes all translations and settings, and when updating it, it resets any unsaved changes to other translations/settings. 

We use a similar narrow update technique for setting updates, see: 

https://github.com/discourse/discourse/blob/main/app/assets/javascripts/admin/addon/components/theme-setting-editor.js#L8-L16

Reply:
Does the above make sense @CvX ?